calico/etcdutils.py

```diff
@@ -310,5 +325,41 @@ def _on_snapshot_loaded(self, etcd_snapshot_response):
         pass
 
 
+def delete_empty_parents(client, child_key, root_key, timeout=DEFAULT_TIMEOUT):
+    """
+    Attempts to delete child_key and any empty parent directories.
+
+    Makes a best effort.  If any of the deletes fail, gives up.  This
+    method is safe, even if another client is writing to the directory (the
+    delete will fail if the directory becomes non-empty).
+
```

Comment:
I'm slightly nervous about this. If you fail to delete a directory, you stop having swallowed the error (but logged), which seems reasonable. But if you fail to delete the child_key object, then you also swallow the error. Is that right? I think that is probably an error that you ought to propagate up.